src/guide/advanced/intro.md

```diff
@@ -161,21 +158,21 @@ opt-level = "z"     # Optimize for size vs speed with "s"/"z"(removes vectorizat
 codegen-units = 1   # Further reduces binary size but increases compilation time
 ```
 
-Here we note a few things; firstly, when you encounter a panic in Rust, it stores a bit of debug information, so that you can interpret what happened inside the program, and debug, even when compiled in `release` mode. We explicitly state that we don't want any of that, and instead want panics to just stop the execution. We're not losing any functionality, despite what you might think, failure to execute a WASM blob isn't considered a warning-level event and is not logged even on the peer, so even if you had some useful information in the panic message, you have no reliable way of retrieving it.
+A few things should be noted; first, Rust stores a lot of debug information(even when compiled in `release` mode) to aid you when you encounter a panic, so that you can debug your application. All this debug information bloats the binary significantly, so instead of storing it, panics are directed to immediately abort the execution. By doing this no crucial functionality is lost, despite to what you might think. Panics are a rare event in the Rust world and the reason of failure to execute a WASM blob isn't logged on the peer, so even if you had some useful information in the panic message, you have no reliable way of retrieving it.
 
-Another step that we've already taken involves working under `no_std`. All of our size-related woes stem from Rust being predominantly statically linked. As such breaking the standard library into more manageable crates, like using `alloc::vec` instead of `std::vec` can help us reduce the size and compilation time[^3].
+Another step that was already taken involves working under `no_std` environment. All of our size-related woes stem from Rust being predominantly statically linked. As such, breaking the standard library into more manageable crates, like using `alloc::vec` instead of `std::vec` can help us reduce the size and compilation time[^3].
```

Comment:
```suggestion
Another step that we've already taken involves working under a  `no_std` environment. All of our size-related woes stem from Rust being predominantly statically linked. As such, breaking the standard library into more manageable crates, like using `alloc::vec` instead of `std::vec` can help us reduce the size and compilation time[^3].
```

We, including the reader. _Was_ in this context implies "inside `iroha_wasm`".